ironfist/src/cpp/combat/creatures.cpp

```diff
@@ -167,4 +205,43 @@ void UnloadCreatures() {
 		if(gArmyNames[i] != NULL) free(gArmyNames[i]);
 		if(gArmyNamesPlural[i] != NULL) free(gArmyNamesPlural[i]);
 	}
+}
+
+void __fastcall GetMonsterCost(int mon, int *const costs) {
+	
+	int i;
+	for (int i = 0; i < NUM_SECONDARY_RESOURCES; i++) {
+		costs[i] = gMonSecondaryResourceCost[mon][i];
+	}
+	costs[RESOURCE_GOLD] = gMonsterDatabase[mon].cost;
+
+	if (!customed_secondary_cost) {

```

Comment:
I think what's left is to merge in master and then replace most of the logic in here with just putting this data into creatures.xml, and then we're good to go!